code/AdminRootController.php

```diff
@@ -36,55 +53,31 @@ public static function get_admin_route()
 
     /**
      * Returns the root admin URL for the site with trailing slash
-     *
-     * @return string
      */
-    public static function admin_url(string $action = '')
+    public static function admin_url(string $action = ''): string
     {
         return Controller::join_links(AdminRootController::get_admin_route(), $action);
     }
 
-    /**
-     * @var string
-     * @config
-     * The LeftAndMain child that will be used as the initial panel to display if none is selected (i.e. if you
-     * visit /admin)
-     */
-    private static $default_panel = SecurityAdmin::class;
-
-    /**
-     * @var array
-     * @internal
-     *
-     * Holds an array of url_pattern => controller k/v pairs, the same as Director::rules. However this is built
-     * dynamically from introspecting on all the classes that derive from LeftAndMain.
-     *
-     * Don't access this directly - always access via the rules() accessor below, which will build this array
-     * the first time it's accessed
-     */
-    private static $adminRules = null;
-
     /**
      * Gets a list of url_pattern => controller k/v pairs for each LeftAndMain derived controller
      */
-    public static function rules()
+    public static function rules(): array
     {
         if (AdminRootController::$adminRules === null) {
             AdminRootController::$adminRules = [];
 
             // Map over the array calling add_rule_for_controller on each
-            $classes = CMSMenu::get_cms_classes(null, true, CMSMenu::URL_PRIORITY);
+            $classes = CMSMenu::get_cms_classes(AdminController::class, true, CMSMenu::URL_PRIORITY);
```

Comment:
`null` defaults to `LeftAndMain` - we want the new controller as the base class.